libgnucash/engine/test/test-guid.cpp

```diff
@@ -65,24 +67,26 @@ run_test (void)
     book = qof_session_get_book (sess);
     do_test ((NULL != book), "book not created");
 
+    std::vector<std::unique_ptr<void,void(*)(void*)>> ents;
     col = qof_book_get_collection (book, "asdf");
     type = qof_collection_get_type (col);
 
-    for (i = 0; i < NENT; i++)
+    for (int i = 0; i < NENT; i++)
     {
-        ent = static_cast<QofInstance*>(g_object_new(QOF_TYPE_INSTANCE, NULL));
         guid_replace(&guid);
-        ent = static_cast<QofInstance*>(g_object_new(QOF_TYPE_INSTANCE,
-                                                     "guid", &guid, NULL));
+        void *go = g_object_new(QOF_TYPE_INSTANCE, "guid", &guid, NULL);
```

Comment:
We need a void* to give to g_object_unref() later on, so I figured it was nicer to store the void* we get from g_object_new() in a separate variable than casting the QofInstance* back to void*.

Reply:
But you could as easily have done
```
struct QofInstanceDeleter {
    operator()(QofInstance* inst) {
         g_object_unref(G_OBJECT(inst));
    }
};

std::vector<std::unique_ptr<QofInstance, QofInstanceDeleter>> ents;
/*…*/
       ents.emplace_back(ent);
```
and it would even be less ugly.